File: app/src/main/java/unforgettable/azcs/me/unforgettable/WordsAdapter.java
package unforgettable.azcs.me.unforgettable;

import android.content.Context;
import android.content.Intent;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import static unforgettable.azcs.me.unforgettable.Utils.WORD;

/**
 * Created by abdulazizalawshan on 12/16/17.
 */

public class WordsAdapter extends RecyclerView.Adapter<WordsAdapter.WordsViewHolder> {

    private List<Word> words = new ArrayList<>();
    private Context context;

    public WordsAdapter(List<Word> words, Context context) {
        this.words = words;
        this.context = context;
    }

    @Override
    public WordsViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new WordsViewHolder(LayoutInflater.from(context).inflate(R.layout.item, parent,false));
    }

    @Override
    public void onBindViewHolder(WordsViewHolder holder, int position) {
        holder.bind(words.get(position));
    }

    @Override
    public int getItemCount() {
        return words.size();
    }

    class WordsViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener{
        TextView mWord, mMeaning;
        Word word;
        public WordsViewHolder(View itemView) {
            super(itemView);
            mWord = itemView.findViewById(R.id.word);
            mMeaning = itemView.findViewById(R.id.meaning);
            itemView.setOnClickListener(this);
        }

        public void bind(Word word) {
            this.word = word;
            mWord.setText(word.getWord());
            mMeaning.setText(word.getMeaning());
        }

        @Override
        public void onClick(View v) {
            Intent intent = new Intent(context,ShowWordActivity.class);
            intent.putExtra(WORD,word);
            context.startActivity(intent);
        }
    }
}

File: app/src/main/java/unforgettable/azcs/me/unforgettable/WordsAdapter.kt
package unforgettable.azcs.me.unforgettable

import android.content.Context
import android.content.Intent
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import unforgettable.azcs.me.unforgettable.Utils.WORD

/**
 * Created by abdulazizalawshan on 12/16/17.
 */

class WordsAdapter(private val context: Context, private val words: List<Word>) : RecyclerView.Adapter<WordsAdapter.WordsViewHolder>() {



    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): WordsViewHolder {
        return WordsViewHolder(LayoutInflater.from(context).inflate(R.layout.item, parent, false))
    }

    override fun onBindViewHolder(holder: WordsViewHolder, position: Int) {
        holder.bind(words[position])
    }

    override fun getItemCount(): Int {
        return words.size
    }

    inner class WordsViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
        private var mWord: TextView = itemView.findViewById(R.id.word)
        private var mMeaning: TextView = itemView.findViewById(R.id.meaning)
        lateinit var word: Word

        init {
            itemView.setOnClickListener(this)
        }

        fun bind(word: Word) {
            this.word = word
            mWord.text = word.word
            mMeaning.text = word.meaning
        }

        override fun onClick(v: View) {
            val intent = Intent(context, ShowWordActivity::class.java)
            intent.putExtra(WORD, word)
            context.startActivity(intent)
        }
    }
}
